src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export interface PostsState {
+  posts: Post[];
+  loaded: boolean;
```

Comment:
```suggestion
  isLoading: boolean;
```

Such variant is more distributed, but up to you